apps/admin/pages/api/auth/register.ts

```diff
@@ -0,0 +1,48 @@
+import { Session } from '@wsvvrijheid/types'
+import { mutation, sessionOptions, getAuth } from '@wsvvrijheid/utils'
+import axios from 'axios'
+import { withIronSessionApiRoute } from 'iron-session/next'
+import { NextApiResponse, NextApiRequest } from 'next'
+
+const registerRoute = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { name, username, email, password } = req.body
+
+  const trimmedName = name.trim()
+  const trimmedUsername = username.trim()
+  const trimmedEmail = email.trim()
+
+  try {
+    const response = await axios.post(
+      `${process.env.NEXT_PUBLIC_API_URL}/api/auth/local/register`,
+      {
+        username: trimmedUsername,
+        email: trimmedEmail,
+        password,
+      },
+    )
+
+    const token = response.data.jwt
+    const userId = response.data.user?.id
+
+    await mutation(token).post('api/artists', {
+      data: { user: userId, name: trimmedName },
+    })
+
+    const auth = await getAuth(email, password)
+
+    req.session = auth as Session
```

Comment:
It's safer to keep session props like `destroy` and `save`

```suggestion
    req.session = { ...auth, ...req.session }
```

Reply:
It's for casting. There is no problem with the other properties of the auth object, but only for the user. Is it safe to not cast at all? Because otherwise there is no casting.

Reply:
I remember req.session object has `save` and `destroy` property functions. I'm just curious if we re assign the session object, will we loose the function properties.

```ts
session = {
  save: () => void
  destroy => void
}
```

Then we try to reassign the session like:

```ts
session = {
  user: User
  token: string
  isLoggedIn: boolean
}
```

That's why I had to do the trick `auth as Session` because typescript was complaining about `save` and `destroy` props which are required props of the session from the library.

We could also do

session.auth = auth or (then adjust the Session type)
session.user = auth.user, session.token = auth.token, session.isLoggedIn = true